File: SpringBoot/PocProject/src/main/java/com/poc/services/OrderService.java
package com.poc.services;

import com.poc.entity.Order;
import com.poc.repository.OrderRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class OrderService {

    @Autowired
    private OrderRepository orderRepository;

    public List<Order> getOrderDetails(){
        return orderRepository.findWideOrders();
    }
}

File: Golang/internal/service/OrderService.go
package service

import (
	"github.com/kamalakshancg/GolangPOC/internal/entity"
	"github.com/kamalakshancg/GolangPOC/internal/repository"
)

type OrderService struct {
	OrderRepo *repository.OrderRepo
}

func NewOrderService(orderRepo *repository.OrderRepo) *OrderService {
	return &OrderService{OrderRepo: orderRepo}
}

func (s *OrderService) GetOrderDetails() ([]entity.Order, error) {
	return s.OrderRepo.GetOrders()
}
